src/mlpack/tests/sumtree_test.cpp

```diff
@@ -13,82 +13,78 @@
 
 #include <mlpack/methods/reinforcement_learning/replay/sumtree.hpp>
 
-#include <boost/test/unit_test.hpp>
-#include "test_tools.hpp"
+#include "catch.hpp"
+#include "test_catch_tools.hpp"
 
 using namespace mlpack;
 using namespace mlpack::rl;
 
-BOOST_AUTO_TEST_SUITE(SumTreeTest);
-
 /**
  * Test that we set the element.
  */
-BOOST_AUTO_TEST_CASE(SetElement)
+TEST_CASE("SetElement", "[SumTreeTest]")
 {
   SumTree<double> sumtree(4);
   sumtree.Set(0, 1.0);
   sumtree.Set(1, 0.8);
   sumtree.Set(2, 0.6);
   sumtree.Set(3, 0.4);
 
-  BOOST_CHECK_CLOSE(sumtree.Sum(), 2.8, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Sum(0, 1), 1.0, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Sum(0, 3), 2.4, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Sum(1, 4), 1.8, 1e-8);
+  CHECK(sumtree.Sum() == Approx(2.8).epsilon(1e-10));
+  CHECK(sumtree.Sum(0, 1) == Approx(1.0).epsilon(1e-10));
+  CHECK(sumtree.Sum(0, 3) == Approx(2.4).epsilon(1e-10));
+  CHECK(sumtree.Sum(1, 4) == Approx(1.8).epsilon(1e-10));
 }
 
 /**
  * Test that we get the element.
  */
-BOOST_AUTO_TEST_CASE(GetElement)
+TEST_CASE("GetElement", "[SumTreeTest]")
 {
   SumTree<double> sumtree(4);
   sumtree.Set(0, 1.0);
   sumtree.Set(1, 0.8);
   sumtree.Set(2, 0.6);
   sumtree.Set(3, 0.4);
 
-  BOOST_CHECK_CLOSE(sumtree.Get(0), 1.0, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Get(1), 0.8, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Get(2), 0.6, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Get(3), 0.4, 1e-8);
+  CHECK(sumtree.Get(0) == Approx(1.0).epsilon(1e-10));
+  CHECK(sumtree.Get(1) == Approx(0.8).epsilon(1e-10));
+  CHECK(sumtree.Get(2) == Approx(0.6).epsilon(1e-10));
+  CHECK(sumtree.Get(3) == Approx(0.4).epsilon(1e-10));
 }
 
 /**
  * Test that we find the highest index in the array such that
  * Sum(arr[0] + arr[1] + arr[2] ... + arr[i]) <= mass.
  */
-BOOST_AUTO_TEST_CASE(FindPrefixSum)
+TEST_CASE("FindPrefixSum", "[SumTreeTest]")
 {
   SumTree<double> sumtree(4);
   sumtree.Set(0, 1.0);
   sumtree.Set(1, 0.8);
   sumtree.Set(2, 0.6);
   sumtree.Set(3, 0.4);
 
-  BOOST_CHECK_EQUAL(sumtree.FindPrefixSum(0), 0);
-  BOOST_CHECK_EQUAL(sumtree.FindPrefixSum(1), 1);
-  BOOST_CHECK_EQUAL(sumtree.FindPrefixSum(2.8), 3);
-  BOOST_CHECK_EQUAL(sumtree.FindPrefixSum(3.0), 3);
+  CHECK(sumtree.FindPrefixSum(0) <= 0);
```

Comment:
Can you check if the style check is fine with `0.0` instead of `0`. Or maybe saving `sumtree.FindPrefixSum(0)` in a temporary `const` parameter works.

Reply:
Hahah , just a false positive

Reply:
Yeah, right now we don't have a mechanism to ignore certain style issue, like we do for the static analysis job, so ideally we can find a workaround here, if not I'll go ahead and add the same mechanism to the style job as well.